### ISSUE TITLE
event.data format sucks

### ISSUE THREAD

**Issue by AzazelHD** [petasis/tkdnd#57]
Hello.

When you drag, lets say 3 files into the TkinterDnD.Tk() window, u get their path, separate with whitespaces.

So what happens if a path contains whitespaces? It is a mess.

Can you separate paths with any other character? like semicolon or something? Or, ideally, return the data on a list, not on a single string.

If no, can u at least tell me what should i modify in order to achieve this?

thanks in advance

**Comment by petasis** [comment]
I am not sure what you mean. During drops, data are passed as a Tcl list. For example:

https://github.com/petasis/tkdnd/blob/master/library/tkdnd_generic.tcl#L388

You are probably referring to python's tkinter. Maybe it is not handled correctly there?

**Comment by AzazelHD** [comment]
Im using tkinterdnd2. Python tkinter doesnt even support dropping files.

```py
import tkinter as tk
from tkinterdnd2 import DND_FILES, TkinterDnD

root = TkinterDnD.Tk()

def handle_drop(event):
    paths = event.data.strip()
    print(paths)


root.geometry("640x400")
root.title("Drag and Drop Example")

drop_zone = tk.Label(root, text="Drop files here", font=("Arial", 30))
drop_zone.pack(expand=True, fill=tk.BOTH)

# register the listbox as a drop target
drop_zone.drop_target_register(DND_FILES)
drop_zone.dnd_bind('<<Drop>>', handle_drop)


root.mainloop()
```

**Comment by AzazelHD** [comment]
I dont know, maybe it is me, but i dont think storing paths on a list is too hard/revolutionary wtf????

**Comment by johnwoltman** [comment]
> I dont know, maybe it is me, but i dont think storing paths on a list is too hard/revolutionary wtf????

I agree, and it would be nice if event.data was a list instead of a weird string.  It's not too bad having to call the `splitlist()` method myself I guess.  The bigger problem is that it would be nice to know that `splitlist `even exists if you (like me) don't know anything about Tcl/Tk under the hood.

I looked at the underlying Tcl stuff in tkinterdnd2 and I do *not* want to learn Tcl, but I might poke around and see what it would take to add the stuff I want to tkinterdnd2.

Sorry, @petasis, for chiming in on your project's bug tracker when it seems more related to tkinterdnd2.

**Comment by eric-gilbertson** [comment]
I too have observed the problems described by johnwalton regarding spaces (and tabs) in file names. The issue is that a file path containing white space is split into multiple partial paths. I worked around the immediate issue by replacing any space characters with an alternate space character such as En Quad, e.g. '\u2000'. It would be very helpful if the path handling could be modified to support file paths with conventional spaces.
